src/engine/structures.rs

```diff
@@ -178,6 +188,14 @@ impl<T: HasName + HasUuid> Table<T> {
             (Some(name_item), Some(uuid_item)) => vec![name_item, uuid_item],
         }
     }
+
+    pub fn iter(&self) -> Iter<T> {
```

Comment:
I think we do not want to be able to mutate the items while they are in the container. The container has an invariant which is tested in the tests. But, essentially, the problem is that there are two keys, and each items also contains its two key values. If it is possible to mutate the items while they are in the table, it is possible to break the invariant, regarding the relationship of the keys to the items they point to.